src/indexer/gitcoin-attesation-network/handleEvent.ts

```diff
@@ -0,0 +1,112 @@
+import { EventHandlerArgs } from "chainsauce";
+import {
+  Hex,
+  decodeAbiParameters,
+} from "viem";
+import { parseAddress } from "../../address.js";
+import { Changeset } from "../../database/index.js";
+import type { Indexer } from "../indexer.js";
+import { AttestationMetadata, GitcoinAttestedData } from "../types.js";
+
+function decodeAttestedData(encodedData: Hex) {
+  const decodedData = decodeAbiParameters(
+    [
+      { name: "projectsContributed", type: "uint64" },
+      { name: "roundsCountributed", type: "uint64" },
+      { name: "chainIdsContributed", type: "uint64" },
+      { name: "totalUSDAmount", type: "uint128" },
+      { name: "timestamp", type: "uint64" },
+      { name: "metadataCid", type: "string" },
+    ],
+    encodedData
+  );
+
+  const results: GitcoinAttestedData = {
+    projectsContributed: decodedData[0],
+    roundsCountributed: decodedData[1],
+    chainIdsContributed: decodedData[2],
+    totalUSDAmount: decodedData[3],
+    timestamp: decodedData[4],
+    metadataCid: decodedData[5],
+  };
+
+  return results;
+}
+
+
+export async function handleEvent(
+  args: EventHandlerArgs<Indexer>
+): Promise<Changeset[]> {
+  const {
+    chainId,
+    event,
+    subscribeToContract,
+    readContract,
+    getBlock,
+    context: {
+      db,
+      rpcClient,
+      ipfsGet,
+      logger,
+      priceProvider,
+      blockTimestampInMs,
+    },
+  } = args;
+
+  switch (event.name) {
+    // -- Allo V2 Profiles
+    case "OnAttested": {
+      const attesationId = event.params.uid;
+      const recipient = parseAddress(event.params.recipient);
+      const fee = event.params.fee;
+      const refUID = event.params.refUID;
+
+      const attesationData = decodeAttestedData(event.params.data);
+
+      let data: AttestationMetadata[] = [];
+      try {
+        data = await ipfsGet(attesationData.metadataCid) ?? [];
+      } catch (e) {
+        logger.warn({
+          msg: `OnAttested: Failed to fetch metadata for attesation ${attesationId}`,
+          event,
+          attesationData,
+        });
+        return [];
+      }    
+
+      let donations = [];
+      for (let i = 0; i < data.length; i++) {
+        const metadata = data[i];
+        donations = await db.getDonationsByTxHash(metadata.chainId, metadata.txnHash);
+        // should we just link the donations table to the attestation?
+        // this means we would only need the txnHash in the attestation metadata
+        // all the other data can be fetched via getDonationsByTxHash
```

Comment:
@0xKurt @hussedev @nijoe1

Reply:
Makes sense. But if others want to use the metadata differently, it might make things harder for them. If that’s not a big deal, we can just include the chainId and transactionHash. Otherwise, we should add everything.